vavr/src/main/java/io/vavr/collection/LinkedHashMap.java

```diff
@@ -701,14 +701,14 @@ public boolean isSequential() {
 
     @Override
     public LinkedHashMap<K, V> put(K key, V value) {
-        Queue<Tuple2<K, V>> newList = list;
-        HashMap<K, V> newMap = map;
-        if (containsKey(key)) {
-            newList = newList.filter(t -> !Objects.equals(t._1, key));
-            newMap = newMap.remove(key);
+        final Queue<Tuple2<K, V>> newList;
+        final Option<V> currentEntry = get(key);
+        if (currentEntry.isDefined()) {
```

Comment:
This is not necessary, we will act like `java.util.HashMap`. It will be important when implementing the `MapView` later.

```java
public class Test {

    public static void main(String[] args) {

        A x = new A(1, 1);
        A y = new A(1, 2);

        // true
        System.out.println(x.equals(y));

        java.util.Map<Integer, A> map = new java.util.HashMap<>();

        map.put(1, x);
        map.put(1, y);

        // true
        System.out.println(map.get(1) == y);

    }

    static class A {

        final int a;
        final int b;

        A(int a, int b) {
            this.a = a;
            this.b = b;
        }

        @Override
        public boolean equals(Object o) {
            if (o == this) {
                return true;
            } else if (o instanceof A) {
                final A that = (A) o;
                return this.a == that.a;
            } else {
                return false;
            }
        }

        @Override
        public int hashCode() {
            return a;
        }

        @Override
        public String toString() {
            return "A(a:" + a + ", b:" + b + ")";
        }

    }
}
```